From 5b925fdc6691fe725a64ee267a6bd247afc69570 Mon Sep 17 00:00:00 2001
From: Thomas Orozco <thomas@orozco.fr>
Date: Wed, 2 Mar 2016 21:49:53 +0100
Subject: [PATCH 1/1] Set authorization header when connecting to ES

Use the authoriation settings from elasticsearch.url when operating as a
proxy.

This mirrors the Kibana 4.1 behavior and makes it possible for users to
use a different password for Kibana and Elasticsearch.
---
 src/plugins/elasticsearch/lib/map_uri.js | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/src/plugins/elasticsearch/lib/map_uri.js b/src/plugins/elasticsearch/lib/map_uri.js
index de78529..988f2d5 100644
--- a/src/plugins/elasticsearch/lib/map_uri.js
+++ b/src/plugins/elasticsearch/lib/map_uri.js
@@ -1,5 +1,6 @@
 var querystring = require('querystring');
-var resolve = require('url').resolve;
+var parseUrl = require('url').parse;
+
 module.exports = function mapUri(server, prefix) {
   var config = server.config();
   return function (request, done) {
@@ -9,8 +10,16 @@ module.exports = function mapUri(server, prefix) {
       if (/\/$/.test(url)) url = url.substring(0, url.length - 1);
       url += path;
     }
+
+    var uri = parseUrl(url),
+        headers = {};
+    if (uri.auth) {
+      var auth = new Buffer(uri.auth);
+      headers.authorization = 'Basic ' + auth.toString('base64');
+    }
+
     var query = querystring.stringify(request.query);
     if (query) url += '?' + query;
-    done(null, url);
+    done(null, url, headers);
   };
 };
-- 
2.7.1

